client/packages/common/src/authentication/hooks/useCallbackWithPermission.ts

```diff
@@ -0,0 +1,17 @@
+import { useAuthContext } from '../AuthContext';
+import { useDisabledNotificationToast } from '@common/components';
+import { UserPermission } from '@common/types';
+
+export const useCallbackWithPermission = (
```

Comment:
this is possibly a little OTT - but I think worth it.
we've used this pattern in a few places now, and rather than repeating, I've consolidated into a hook

Reply:
Nice, I almost wonder if we should call it `useCallbackIfPermission` or something? 
Can't think of a really nice clear name, I think it's ok how it is. 

At least it has Permission in the name rather than `useDisabledNotificationToast` which doesn't obviously relate to permissions...

Reply:
funnily enough.. `useCallbackIfPermitted` was my first name attempt and then I dithered and came up with this. Naming stuff.. the eternal problem! Agree though - having 'permission' in there is clearer than 'disabled notification' is!